src/data/Data.js

```diff
@@ -14,34 +14,25 @@ Data.accepts = jQuery.acceptData;
 Data.prototype = {
 
 	register: function( owner, initial ) {
-		var descriptor = {},
+		var descriptor,
 			value = initial || {};
 
-		try {
-			// If it is a node unlikely to be stringify-ed or looped over
-			// use plain assignment
-			if ( owner.nodeType ) {
-				owner[ this.expando ] = value;
+		// If it is a node unlikely to be stringify-ed or looped over
+		// use plain assignment
+		if ( owner.nodeType ) {
+			owner[ this.expando ] = value;
 
-			// Otherwise secure it in a non-enumerable, non-writable property
-			// configurability must be true to allow the property to be
-			// deleted with the delete operator
-			} else {
-				descriptor[ this.expando ] = {
-					value: value,
-					writable: true,
-					configurable: true
-				};
-				Object.defineProperties( owner, descriptor );
-			}
-
-		// Support: Android < 4
-		// Fallback to a less secure definition
-		} catch ( e ) {
-			descriptor[ this.expando ] = value;
-			jQuery.extend( owner, descriptor );
+		// Otherwise secure it in a non-enumerable, non-writable property
+		// configurability must be true to allow the property to be
+		// deleted with the delete operator
+		} else {
+			descriptor = {
```

Comment:
Is there a reason to keep the variable? Why not just
```javascript
Object.defineProperty( owner, this.expando, {
	value: value,
	writable: true,
	configurable: true
});
```
(which I see in a few other places in the code, not sure what the convention is though...)